platzky_redirections/plugin.py

```
@@ -0,0 +1,99 @@
+def setup_routes(app: Engine, redirections: list[Redirection]) -> None:
+    """
+    Set up Flask routes for redirections.
+
+    Args:
+        app: Flask application instance
+        redirections: List of Redirection objects
+
+    Raises:
+        ValueError: If route conflicts are detected
+    """
+    existing_routes: set[str] = set(rule.rule for rule in app.url_map.iter_rules())
```

Ruff (0.15.0)

[convention] Unnecessary generator (rewrite as a set comprehension)

Rewrite as a set comprehension

(C401)